packages/format-library/src/link/modal.native.js

```diff
@@ -7,6 +7,8 @@ import { Platform } from 'react-native';
 /**
  * WordPress dependencies
  */
+import { compose } from '@wordpress/compose';
```

Comment:
I am not sure changing this component for Button block is the best thing to do as it introduced so much complexity here. And after the latest commit the clipboard link is not working anymore for all RichText instanced, so it has side effects too.

As far as I understand this component is working based on the format attributes rich text has, but in case of a Button block we don't have those, so, the inputs are actually different. As well as the actions we want to do afterwards.

If possible I'd want to remove RichText's link button from the toolbar and add our own link button, then, show a link modal manually for Button block. I know we want to reuse some logic but can we consider some other methods for sharing code? Some options that pops into my mind:


1. Maybe we could extract out the UI parts(I mean the part inside render function) and create a new modal for Button component, reusing the UI components only.

2. Another option: I know inheritance is not very Javascripty but we can consider it to make minimum changes on the existing link/modal.native.js and override some methods to change functionality for Button block. 

I am also open for other ideas.

@koke please drop some thoughts if you feel like there's a better way.

Reply:
Unfortunately, syncing link options and button options required these specific changes which I am not proud of. Actually, I'm a bit torn between keeping link icon for `Button` and removing that (all the same functionality is in button settings). I understand that removing icon results in losing some kind of consistency, but on the other hand different settings in link options are also similar to lack consistency.

Reply:
Last commits added custom `link settings` adjusted for `Button` component